### PR TITLE
feat/use monaco: TypeScript fixups

### PR DESCRIPTION
**Pull request by cshaver**
To be merged into https://github.com/acao/graphiql/pull/17.

Doing some Sunday afternoon typescript fixes. I was curious about a couple `@ts-ignore`s and got carried away, so I tried to resolve all the ones I could, plus a couple spelling edits and year updates along the way.

### CODE REVIEW

File: packages/graphiql/src/types.d.ts [hunk 1]
```diff
@@ -1,6 +0,0 @@
-import { GraphQLType } from 'graphql';
-
-export namespace GraphiQL {
-  export type GetDefaultFieldNamesFn = (type: GraphQLType) => string[];
-  export type Maybe<T> = T | null | undefined;
-}
```
Comment (cshaver) [inline packages/graphiql/src/types.d.ts:6]: Whoops, I think I need to put this back?

File: packages/monaco-graphql/src/typings/monaco.d.ts [hunk 2]
```diff
@@ -5,118 +5,111 @@
  *  LICENSE file in the root directory of this source tree.
  */
```
Comment (cshaver) [inline packages/monaco-graphql/src/typings/monaco.d.ts:7]: [Hide whitespace changes for this file](https://github.com/acao/graphiql/pull/18/files?diff=split&w=1#diff-93080d2ce6a8ca3bfb9e955bbadbcc1e)

File: packages/graphiql/src/components/GraphiQL.tsx [hunk 3]
```diff
@@ -453,43 +453,42 @@ class GraphiQLInternals extends React.Component<
...
-            if (index < cursorIndex) {
-              newCursorIndex += string.length;
-            }
-          });
-          editor.setCursor(editor.posFromIndex(newCursorIndex));
-        });
-      }
-    }
-    return result;
+    // const { insertions, result } = fillLeafs(
```
Comment (acao) [inline packages/graphiql/src/components/GraphiQL.tsx:456]: ah yes! i think this is where we get the auto-expanding fields? trying to figure out how to re-enable that in monaco today

**Comment by acao** [comment]
@cshaver I can port over a good portion of these changes to the current RFC branch in a PR, there's some very useful improvements here!

File: packages/monaco-graphql/src/utils.ts [hunk 4]
```diff
@@ -11,13 +11,13 @@ import type {
   Diagnostic,
   CompletionItem as GraphQLCompletionItem,
 } from 'graphql-language-service-types';
+import type monaco from 'monaco-editor';
```
Comment (acao) [inline packages/monaco-graphql/src/utils.ts:14]: it's available globally across the package now, so no need for this! i just need to add it to eslint globals

**Comment by acao** [comment]
@cshaver I think I captured most of these improvements, thank you!

**Pull request closed without merging**